alfasim_sdk_api/detail/bootstrap_win.h

```diff
@@ -5,94 +5,89 @@
-        get_layer_id = (get_layer_id_func)GetProcAddress(this->dllHandle, "get_layer_id");
-        get_state_variable_array = (get_state_variable_array_func)GetProcAddress(this->dllHandle, "get_state_variable_array");
-        get_simulation_array = (get_simulation_array_func)GetProcAddress(this->dllHandle, "get_simulation_array");
-        get_simulation_tracer_array = (get_simulation_tracer_array_func)GetProcAddress(this->dllHandle, "get_simulation_tracer_array");
-        get_simulation_quantity = (get_simulation_quantity_func)GetProcAddress(this->dllHandle, "get_simulation_quantity");
-        get_wall_interfaces_temperature = (get_wall_interfaces_temperature_func)GetProcAddress(this->dllHandle, "get_wall_interfaces_temperature");
-        get_flow_pattern = (get_flow_pattern_func)GetProcAddress(this->dllHandle, "get_flow_pattern");
-        get_plugin_input_data_table_quantity = (get_plugin_input_data_table_quantity_func)GetProcAddress(this->dllHandle, "get_plugin_input_data_table_quantity");
-        get_tracer_id = (get_tracer_id_func)GetProcAddress(this->dllHandle, "get_tracer_id");
-        get_tracer_name_size = (get_tracer_name_size_func)GetProcAddress(this->dllHandle, "get_tracer_name_size");
-        get_tracer_name = (get_tracer_name_func)GetProcAddress(this->dllHandle, "get_tracer_name");
-        get_tracer_ref_by_name = (get_tracer_ref_by_name_func)GetProcAddress(this->dllHandle, "get_tracer_ref_by_name");
-        get_tracer_partition_coefficient = (get_tracer_partition_coefficient_func)GetProcAddress(this->dllHandle, "get_tracer_partition_coefficient");
-        get_wall_layer_id = (get_wall_layer_id_func)GetProcAddress(this->dllHandle, "get_wall_layer_id");
-        set_wall_layer_property = (set_wall_layer_property_func)GetProcAddress(this->dllHandle, "set_wall_layer_property");
-        get_plugin_input_data_multiplereference_selected_size = (get_plugin_input_data_multiplereference_selected_size_func)GetProcAddress(this->dllHandle, "get_plugin_input_data_multiplereference_selected_size");
+    // Extract the folder from the environment variable
+    WCHAR *executable_dir_from_env = NULL;
+    size_t _len = 0;
+    _wdupenv_s(&executable_dir_from_env, &_len, L"ALFASIM_PATH");
 
-        free(executable_dir_from_env);
+    // Find the alfasim installation path: Prefer the environment variable ALFASIM_PATH, but, if not found,
+    // try the same path as the executable
+    WCHAR *alfasim_executable_dir;
+    if (executable_dir_from_env != NULL) {
+        alfasim_executable_dir = executable_dir_from_env;
+    } else {
+        alfasim_executable_dir = current_exe_dir;
     }
 
-    ~alfasim_sdk_bootstrap()
-    {
-        FreeLibrary(this->dllHandle);
-    }
+    // Load shared object
+    WCHAR full_filepath[MAX_PATH];
+    full_filepath[0] = '\0';
+    StringCchCatW(full_filepath, MAX_PATH, alfasim_executable_dir);
+    StringCchCatW(full_filepath, MAX_PATH, DLL_FILENAME);
```

Comment:
Consider printing explicit errors when the returned strings would be larger than `MAX_PATH` (sometimes paths in CI are very large, it's not a very common error, but it may happen)